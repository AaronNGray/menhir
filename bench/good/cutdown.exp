Warning: one state has shift/reduce conflicts.
Warning: one state has reduce/reduce conflicts.
Warning: one shift/reduce conflict was arbitrarily resolved.
Warning: one reduce/reduce conflict was arbitrarily resolved.
