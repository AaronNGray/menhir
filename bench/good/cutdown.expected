%{ %}
%start goal
%token R_BRACKET
%token NEW
%token L_BRACKET
%token <string> INTEGER_LITERAL
%token EOF
%token ASSIGN

%type <unit> goal
%%

goal:
| assignment EOF
    { }

primary:
| INTEGER_LITERAL
    { }
| array_access
    { }

primary_not_name:
| array_access
    { }
| NEW L_BRACKET expression R_BRACKET
    { }

array_access:
| expression L_BRACKET primary_not_name R_BRACKET
    { }

expression:
| primary
    { }
| assignment
    { }

assignment:
| left_hand_side ASSIGN expression
    { }

left_hand_side:
| primary_not_name
    { }

%%



